bus/services/Makefile.am

```diff
@@ -0,0 +1,68 @@
+# vim:set noet ts=4:
+#
+# ibus - The Input Bus
+#
+# Copyright (c) 2022 Takao Fujiwara <takao.fujiwara1@gmail.com>
+# Copyright (c) 2022 Red Hat, Inc.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+# USA
+
+NULL =
+
+# Originally this file was added for GNOME but seems not to be used
+# at present.
+#
+# TODO: Check if flatpack uses this file and we could rename
+# org.freedesktop.IBus.service to org.freedesktop.IBus.DBus.GNOME.service
+dbusservice_in_files = org.freedesktop.IBus.service.in
+dbusservice_DATA = $(dbusservice_in_files:.service.in=.service)
+dbusservicedir=${datadir}/dbus-1/services
```

Comment:
The D-Bus service file should continue to be be named `${bus_name}.service`, where `${bus_name}` is the D-Bus well-known name that ibus requests (with `g_bus_own_name()` or similar, the same thing as the `BusName` in the systemd service files). If the D-Bus well-known name does not change, then it would be incorrect to rename it.

If you want ibus to continue to work by being launched by `dbus-daemon` on systems that do not use `systemd --user` or do not run the session bus as a systemd user service, then it still needs this D-Bus `.service` file.

In particular, the dbus-daemon cannot be run as a systemd user service on any system that wants to be able to run multiple independent sessions as the same uid, each with its own session bus (such as the gdm greeter and some VNC/RFB remote-access configurations).

Obviously the dbus-daemon also cannot be run as a systemd user service on systems that use some other init system instead of systemd. I am in favour of improving the situation for users of systemd (like me), but not at the expense of making it impossible for people who do not use systemd to use ibus.

Reply:
Thank you for the review. I updated my comment.